tests/base.py

```diff
@@ -330,6 +332,89 @@ def setUp(self):
         super(ApiTestCase, self).setUp()
         settings.USE_EMAIL = False
 
+class ApiAddonTestCase(ApiTestCase):
+    """Base `TestCase` for tests that require interaction with addons.
+
+    """
+
+    @abc.abstractproperty
+    def short_name(self):
+        pass
+
+    @abc.abstractproperty
+    def addon_type(self):
+        pass
+
+    @abc.abstractmethod
+    def _apply_auth_configuration(self):
+        pass
+
+    @abc.abstractmethod
+    def _set_urls(self):
+        pass
+
+    def _settings_kwargs(self, node, user_settings):
+        return {
+            'user_settings': self.user_settings,
+            'folder_id': '1234567890',
+            'owner': self.node
+        }
+
+    def setUp(self):
+        super(ApiAddonTestCase, self).setUp()
+        from tests.factories import (
+            ProjectFactory,
+            AuthUserFactory,
+        )
+        from website.addons.base import (
+            AddonOAuthNodeSettingsBase, AddonNodeSettingsBase,
+            AddonOAuthUserSettingsBase, AddonUserSettingsBase
+        )
+        # TODO(mfraezz:) handle type-specific setUp for each type individually
+        assert self.addon_type in ('CONFIGURABLE', 'OAUTH', 'MANAGEABLE', 'UNMANAGEABLE', 'INVALID')  
```

Comment:
Some of these options seem like they could overlap. Are they test-specific, or should the allowed values be documented somewhere?

(I'm trying to flag anything that seems like it could lead to mystery schemas)